jmx-scraper/src/test/java/io/opentelemetry/contrib/jmxscraper/config/JmxScraperConfigFactoryTest.java

```diff
@@ -0,0 +1,325 @@
+            "otel.jmx.custom.jmx.scraping.config or otel.jmx.target.system must be specified.");
+  }
+
+  @Test
+  void shouldFailValidation_invalidTargetSystem() {
+    // Given
+    JmxScraperConfigFactory configFactory = new JmxScraperConfigFactory();
+    Properties properties = (Properties) validProperties.clone();
+    properties.setProperty(JmxScraperConfigFactory.TARGET_SYSTEM, "hal9000");
+    JmxScraperConfig config = configFactory.createConfig(properties);
+
+    // When and Then
+    assertThatThrownBy(() -> configFactory.validateConfig(config))
+        .isInstanceOf(ConfigurationException.class)
+        .hasMessage(
+            "[hal9000] must specify targets from "
+                + JmxScraperConfigFactory.AVAILABLE_TARGET_SYSTEMS);
+  }
+
+  @Test
+  void shouldFailValidation_missingOtlpEndpoint() {
+    // Given
+    JmxScraperConfigFactory configFactory = new JmxScraperConfigFactory();
+    Properties properties = (Properties) validProperties.clone();
+    properties.remove(JmxScraperConfigFactory.OTLP_ENDPOINT);
+    properties.setProperty(JmxScraperConfigFactory.METRICS_EXPORTER_TYPE, "otlp");
+    JmxScraperConfig config = configFactory.createConfig(properties);
+
+    // When and Then
+    assertThatThrownBy(() -> configFactory.validateConfig(config))
+        .isInstanceOf(ConfigurationException.class)
+        .hasMessage("otel.exporter.otlp.endpoint must be specified for otlp format.");
+  }
+
+  @Test
+  void shouldFailValidation_negativeInterval() {
+    // Given
+    JmxScraperConfigFactory configFactory = new JmxScraperConfigFactory();
+    Properties properties = (Properties) validProperties.clone();
+    properties.setProperty(JmxScraperConfigFactory.INTERVAL_MILLISECONDS, "-1");
+    JmxScraperConfig config = configFactory.createConfig(properties);
+
+    // When and Then
+    assertThatThrownBy(() -> configFactory.validateConfig(config))
+        .isInstanceOf(ConfigurationException.class)
+        .hasMessage("otel.jmx.interval.milliseconds must be positive.");
+  }
+
+  @Test
+  void shouldFailConfigCreation_invalidInterval() {
+    // Given
+    JmxScraperConfigFactory configFactory = new JmxScraperConfigFactory();
+    Properties properties = (Properties) validProperties.clone();
+    properties.setProperty(JmxScraperConfigFactory.INTERVAL_MILLISECONDS, "abc");
+
+    // When and Then
+    assertThatThrownBy(() -> configFactory.createConfig(properties))
+        .isInstanceOf(ConfigurationException.class)
+        .hasMessage("Failed to parse otel.jmx.interval.milliseconds");
+  }
+
+  //  @ClearSystemProperty(key = "otel.metric.export.interval")
+
+  //  @Test
+  //  @SetSystemProperty(key = "otel.jmx.service.url", value = "myServiceUrl")
+  //  @SetSystemProperty(key = "otel.jmx.groovy.script", value = "myGroovyScript")
+  //  @SetSystemProperty(
+  //      key = "otel.jmx.target.system",
+  //      value = "mytargetsystem,mytargetsystem,myothertargetsystem,myadditionaltargetsystem")
```

Comment:
Is it relevant to merge those commented lines right now or are they just a reminder of the extra tests that are left to implement ?

Reply:
Work on this test suite is not finished yet. I wanted to keep it until I'm done. Final code will not heave it.